ResStock-HPXML: timeseries end convention

joseph-robertson opened the pull request:
## Pull Request Description

Adds support for new `timeseries_timestamp_convention` argument of the ReportSimulationOutput measure. See https://github.com/NREL/resstock/pull/999 for context.

## Checklist

Not all may apply

- [x] Code changes (must work)
- [ ] Tests exercising your feature/bug fix (check coverage report on Checks -> BuildStockBatch Tests -> Artifacts)
- [ ] Coverage has increased or at least not decreased. Update `minimum_coverage` in `.github/workflows/ci.yml` as necessary.
- [ ] All other unit tests passing
- [ ] Update validation for project config yaml file changes
- [ ] Update existing documentation
- [ ] Run a small batch run to make sure it all works (local is fine, unless an Eagle specific feature)
- [x] Add to the changelog_dev.rst file and propose migration text in the pull request

joseph-robertson committed:
Add default end ts convention to hpxml wfg.

joseph-robertson committed:
Update hpxml wfg test.

afontani commented:
@joseph-robertson: How does this relate to this PR in buildstockbatch: https://github.com/NREL/buildstockbatch/pull/301

joseph-robertson commented:
> @joseph-robertson: How does this relate to this PR in buildstockbatch: #301

https://github.com/NREL/OpenStudio-HPXML/pull/1198 changed default to start of timestamp convention. This PR will maintain (existing) behavior of the end of timestamp convention.

So I _believe_ instead of #301 you'd be able to control timestamp convention from the yml.

joseph-robertson committed:
Merge branch 'develop' into ts-convention

joseph-robertson commented:
@afontani Confimed. We can use `timeseries_timestamp_convention` set to `start` (instead of default `end`) to achieve what #301 does: https://github.com/NREL/resstock/pull/999/commits/965b1a4bc1faa7912aff6b04d3188066497e73ce. Supports ResStock, not ComStock.

joseph-robertson committed:
Update docs. [ci skip]

joseph-robertson committed:
Add ts arg to hpxml wfg spec.

afontani commented on buildstockbatch/workflow_generator/residential_hpxml.py:
```diff
@@ -315,6 +316,7 @@ def create_osw(self, sim_id, building_id, upgrade_idx):
             'include_timeseries_zone_temperatures': False,
             'include_timeseries_airflows': False,
             'include_timeseries_weather': False,
+            'timeseries_timestamp_convention': 'end',
```
So it looks like the new YAML argument is not required and defaulted to end. This would maintain the current behavior which is good.

afontani commented:
@joseph-robertson : 
- Do we need to update the changelog [docs/changelog/changelog_dev.rst](https://github.com/NREL/buildstockbatch/tree/develop/docs/changelog/changelog_dev.rst)?
- Is this the corresponding [resstock PR #999](https://github.com/NREL/resstock/pull/999) that needs to be merged along with this PR?

joseph-robertson committed:
Update the dev changelog.

joseph-robertson commented:
> @joseph-robertson :
> 
> * Do we need to update the changelog [docs/changelog/changelog_dev.rst](https://github.com/NREL/buildstockbatch/tree/develop/docs/changelog/changelog_dev.rst)? I updated it.
> * Is this the corresponding [resstock PR #999](https://github.com/NREL/resstock/pull/999) that needs to be merged along with this PR? Correct. I updated the PR description.

joseph-robertson committed:
Merge branch 'develop' into ts-convention

joseph-robertson commented:
Where did we land with this? Can we get this merged?

afontani commented:
@joseph-robertson : I am good to merge this, but I am less aware of what needs to be done for the checklist in this repo. 

@nmerket are you okay to merge this?
